test/images/fakegitserver/Makefile

```diff
@@ -12,21 +12,17 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-.PHONY:	all image push clean
+BIN=fakegitserver
+ARCH ?= amd64
+TARGET ?= $(CURDIR)
+GOLANG_VERSION ?= latest
 
-TAG = 0.1
-PREFIX = gcr.io/google_containers
-IMAGE = fakegitserver
-ARCH = amd64
+bin:
+	docker run -it -v $(TARGET):$(TARGET):Z -v `pwd`/../../../:/go/src/k8s.io/kubernetes:Z \
+	golang:$(GOLANG_VERSION) \
+	/bin/bash -c "\
+		cd /go/src/k8s.io/kubernetes/test/images/fakegitserver && \
+		CGO_ENABLED=0 GOARM=$(GOARM) GOARCH=$(ARCH) go build -a -installsuffix cgo --ldflags '-w' -o $(TARGET)/$(BIN) ."
+	@echo $(shell git rev-parse HEAD) > $(TARGET)/GITHASH.txt
 
-image:
-	./prepare.sh $(ARCH)
-	docker build --pull -t $(PREFIX)/$(IMAGE):$(TAG) .
-
-push: image
-	gcloud docker -- push $(PREFIX)/$(IMAGE):$(TAG)
-
-all: push
-
-clean:
-	rm -f gitserver GITHASH.txt
+.PHONY: bin
```

Comment:
This I've added based on review comment by @ixdy https://github.com/kubernetes/kubernetes/pull/44910#discussion_r123680354